Adds a copy-url-button component

mynameistechno opened the pull request:
This PR:
* Adds a copy-url-button component that leverages `react-copy-to-clipboard`
* Adds the button to components list, inspect, and curate pages (not sure we want them on all, but can easily change)
* A couple UX clean up items (just experimenting, I can revert/change):
  * Groups the icon buttons in the component list
  * Moves the badge over the left so it doesn't shift the buttons to the left after it renders (this is temporary, long term we can do better)
  * Adds bootstrap's "well" styling around the filter boxes

![image](https://user-images.githubusercontent.com/299129/36719791-f2d1a5a4-1b5a-11e8-9be2-48a92b767692.png)

On the Inspect/Curate pages I tried a more subtle "button link" style floated to the right. I didn't want the user to confuse it with a search button (but it looks good as a button and I'm open to changing it or removing altogether):

![image](https://user-images.githubusercontent.com/299129/36720004-84d7b4de-1b5b-11e8-9983-fe81fe2c79d1.png)

jeffmcaffer commented on src/components/CopyUrlButton.js:
```diff
@@ -0,0 +1,66 @@
+  }
+
+  render() {
+    const {path, bsStyle = 'link', className} = this.props
```
suggest adding the `bsStyle` default value as a default prop to make it clear to component users.

jeffmcaffer commented on src/components/CopyUrlButton.js:
```diff
@@ -0,0 +1,66 @@
+export default class CopyUrlButton extends Component {
+
+  static propTypes = {
+    path: PropTypes.string
```
should have bsStyle and className as declared props (with defaults ideally)

mynameistechno replied:
Sounds good, will define and add defaults for all props.

jeffmcaffer commented on src/components/CopyUrlButton.js:
```diff
@@ -0,0 +1,66 @@
+  onCopy() {
+    this.state.timeoutId && clearTimeout(this.state.timeoutId)
+    const timeoutId = setTimeout(this.didCopy, 2000)
+    this.setState({...this.state, copied: true, timeoutId})
```
formatting nit wrt space after { and before } for object literals.

mynameistechno replied:
Prettier wherefore art thou 😜

jeffmcaffer commented on src/components/PageComponentDetails.js:
```diff
@@ -118,10 +118,13 @@ class PageInspect extends Component {
       <Grid className='main-container'>
-        <Row className="show-grid spacer">
-          <Col md={10} mdOffset={1}>
+        <Row className="show-grid well">
```
not keen on this being a well. The design is meant to be super clean without boxes, borders, grid lines, ...  Is this needed?

mynameistechno replied:
No I can remove, it's just that `spacer` had issues. I'll just add more margin between navbar and content to achieve same effect as `spacer`.

mynameistechno commented:
Feedback addressed @jeffmcaffer. Also, I removed the `well` class and did some minor cleanup in the components list. The source line/sha was causing wrapping on the buttons on the right, so I added ellipsis text overflow support for the source value so the layout handles large source values more gracefully (and a couple other css fixes):

Before:
![image](https://user-images.githubusercontent.com/299129/36864208-e9b48b8e-1d40-11e8-9448-6adb515131b7.png)

After:
![image](https://user-images.githubusercontent.com/299129/36864693-3f66d19e-1d42-11e8-81ce-0f5abda6dec8.png)

The panel/list expand below needs some work, but that can be tackled separately. E.g. if browser width is narrowed a bit:
![image](https://user-images.githubusercontent.com/299129/36864749-6a7e6612-1d42-11e8-96bd-fca2fd5f627b.png)

Created #74 to that effect.

We should prob merge #68 first so I can make sure the filterbar renders well with the facets.
